src/swap/handle_swap_sign_transaction.c

```diff
@@ -119,7 +121,75 @@ bool swap_copy_transaction_parameters(create_transaction_parameters_t* params) {
     return true;
 }
 
-bool swap_check_validity() {
+static address_t* swap_get_tx_recipient_address(bool is_jetton_swap) {
+    address_t* recipient = NULL;
+
+    /*
+     *  XXX:
+     *   In case of Jetton transfer, transaction `to` address is the sender jetton wallet address.
+     *   Recipient address is the second hints in message hints.
+     */
+    if (is_jetton_swap) {
+        if (G_context.tx_info.transaction.hints.hints_count >= 2) {
```

Comment:
I think this line should be removed no ?

Reply:
Agreed